### PR TITLE
Story/quad 63

### PR DESCRIPTION
**Pull request by ajaymodi**
We have changed code according to suggestion with proper formatting. Please review and merge.

### CODE REVIEW

**Comment by jdamerow** [comment]
please resolve conflicts

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java [hunk 1]
```diff
@@ -448,5 +465,60 @@
...
+                        throw new QuadrigaException(e.getMessage(), e);
+                    }
+
+                }
+            }
+            
+            return new ResponseEntity<String>(ja.toString(), HttpStatus.OK);
+            
+        } catch (QuadrigaStorageException e) {
+            throw new QuadrigaStorageException();
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java:521]: the try/catch block should be where the exception is actually thrown. If one collection can't be retrieved but the others can, you want to show the the once that you could retrieve and just show an error. Also, given that this is a method that returns JSON, return proper error codes and useful error messages. And don't swallow the exception ;)

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java [hunk 2]
```diff
@@ -193,7 +212,7 @@
      * @param dictionaryId
      * @param model
      * @return
-     * @throws RestException
+     * @throws Exception
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:215]: why did you change that?

Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:215]: and the ones above?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java [hunk 3]
```diff
@@ -404,5 +423,61 @@
...
+            QuadrigaException {
+        String userId = principal.getName();
+        List<IProjectDictionary> dictionaryList = null;
+        try {
+            // TODO: listProjectDictionary() is to be changed according to
+            // mapper
+            dictionaryList = projectDictionaryManager.listProjectDictionary(
+                    projectid, userId);
+            
+            dictionaryList.removeAll(Collections.singleton(null));
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:457]: why are you doing this? there shouldn't be any null in the list, or did you have that case?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java [hunk 4]
```diff
@@ -404,5 +423,61 @@
...
+                        throw new QuadrigaException(e.getMessage(), e);
+                    }
+                }
+            }
+                 
+            return new ResponseEntity<String>(ja.toString(), HttpStatus.OK);
+
+        } catch (QuadrigaStorageException e) {
+            throw new QuadrigaStorageException();
+        }
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:478]: same as before, try/catch needs to be where it's actually thrown. and proper error handling is needed.

Comment (ajaymodi) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:478]: So when you say, proper error handling should we just output status code like 200, 404 or you still want it to have quadrigaexception?

Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:478]: yes, return have it return something that can be used by the client, with the correct status code and  an error message. In other words, catch the exception and return a ResponseEntity.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java (hunk 3 above)
Comment (ajaymodi) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:457]: This code was already there in dictionarycontroller, we are just creating Json Object and returning it.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java (hunk 2 above)
Comment (ajaymodi) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:215]: This was a mistake from our side due to lot of merge conflicts. Let me know if we still miss anything.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java (hunk 3 above)
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:457]: just remove it unless you get errors when you remove it, but I don't think we need this. If it's necessary the underlying code should be changed to make sure the list doesn't have any nulls in it.

**Comment by jdamerow** [comment]
please resolve conflicts

**Comment by ajaymodi** [comment]
please check pull request from story quad-62. We have merged 63 into 62.
